tests/openapi/client/openapi_task_check

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+import sys
+import requests
+
+NO_SHARD_TASK_NAME = "test-no-shard"
+SHARD_TASK_NAME = "test-shard"
+SOURCE1_NAME = "mysql-01"
+SOURCE2_NAME = "mysql-02"
+
+
+API_ENDPOINT = "http://127.0.0.1:8361/api/v1/tasks"
+
+
+def start_task_failed():
+    task = {
+        "name": "test",
+        "task_mode": "all",
+        "shard_mode": "pessimistic_xxd",  # pessimistic_xxd is not a valid shard mode
+        "meta_schema": "dm-meta",
+        "enhance_online_schema_change": True,
+        "on_duplicate": "error",
+        "target_config": {
+            "host": "127.0.0.1",
+            "port": 4000,
+            "user": "root",
+            "password": "",
+        },
+        "table_migrate_rule": [
+            {
+                "source": {
+                    "source_name": SOURCE1_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": "t"},
+            },
+            {
+                "source": {
+                    "source_name": SOURCE2_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": "t"},
+            },
+        ],
+        "source_config": {
+            "source_conf": [
+                {"source_name": SOURCE1_NAME},
+                {"source_name": SOURCE2_NAME},
+            ],
+        },
+    }
+    resp = requests.post(url=API_ENDPOINT, json={"remove_meta": True, "task": task})
+    print("start_task_failed resp=", resp.json())
+    assert resp.status_code == 400
+
+
+def start_noshard_task_success():
+    task = {
+        "name": NO_SHARD_TASK_NAME,
+        "task_mode": "all",
+        "shard_mode": "pessimistic",
+        "meta_schema": "dm-meta",
+        "enhance_online_schema_change": True,
+        "on_duplicate": "error",
+        "target_config": {
+            "host": "127.0.0.1",
+            "port": 4000,
+            "user": "root",
+            "password": "",
+        },
+        "table_migrate_rule": [
+            {
+                "source": {
+                    "source_name": SOURCE1_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": ""},
```

Comment:
can we omit this field if no routing?

Reply:
no, because i use required `table` in spec and i think users are less likely to misunderstand the configuration.

Reply:
not sure if the `target.table` should be "" or "*" cc @sunzhaoyang 

https://docs.oracle.com/goldengate/1212/gg-winux/GWURF/gg_parameters160.htm#CEGBBCHF

Reply:
@sunzhaoyang also think [*] is better, fixed in https://github.com/pingcap/dm/pull/2141/commits/62bd38583921f1efef5979229e1dace9450348f9